tests/concepts_test.py

```diff
@@ -0,0 +1,79 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2022 Scipp contributors (https://github.com/scipp)
+# @author Simon Heybrock
+import scipp as sc
+from scipp.core import concepts
+
+
+def test_concrete_dims_given_single_dim_returns_dim_as_tuple():
+    var = sc.empty(dims=('xx', 'yy'), shape=(2, 2))
+    assert concepts.concrete_dims(var, 'xx') == ('xx', )
+
+
+def test_concrete_dims_given_dim_list_returns_dim_tuple():
+    var = sc.empty(dims=('xx', 'yy'), shape=(2, 2))
+    assert concepts.concrete_dims(var, ['yy', 'xx']) == ('yy', 'xx')
+
+
+def test_concrete_dims_given_none_returns_obj_dims():
+    var = sc.empty(dims=('xx', 'yy'), shape=(2, 2))
+    assert concepts.concrete_dims(var, None) == ('xx', 'yy')
+
+
+def test_irreducible_mask_returns_union_of_relevant_masks():
+    da = sc.DataArray(sc.empty(dims=('xx', 'yy'), shape=(2, 3)))
+    da.masks['x'] = sc.array(dims=['xx'], values=[False, True])
+    da.masks['y'] = sc.array(dims=['yy'], values=[False, True, False])
+    da.masks['xy'] = sc.array(dims=['xx', 'yy'],
+                              values=[[False, False, False], [False, False, True]])
+    assert sc.identical(concepts.irreducible_mask(da, 'xx'),
+                        da.masks['x'] | da.masks['xy'])
+    assert sc.identical(concepts.irreducible_mask(da, 'yy'),
+                        da.masks['y'] | da.masks['xy'])
+    assert sc.identical(concepts.irreducible_mask(da, ('xx', 'yy')),
+                        da.masks['x'] | da.masks['y'] | da.masks['xy'])
+
+
+def test_irreducible_mask_returns_None_if_all_masks_unrelated():
```

Comment:
From the test name, I would have thought that there would be at least two masks.

Reply:
> As we discussed, it would be useful to have some drawings somewhere describing the mechanism. In the docs, they would be quickly out of date. It might be difficult to add in the code.
> Maybe just in the PR description? This would at least not become out of date for the changes implemented in the PR?

Which part would you like to have a drawing for? Note that the mechanism is now considerably simplified compared to what I showed you originally.